src/analysis/calculus/times_cont_diff.lean

```diff
@@ -14,591 +14,1158 @@ By induction, it is `C^n` if it is `C^{n-1}` and its (n-1)-th derivative is `C^1
-structure. This can certainly be done, but I decided it was not worth it for now. Therefore, the
-definition is only made when E and F live in the same universe.
-
-Regarding the definition of `C^n` functions, there are two equivalent definitions:
-* require by induction that the function is differentiable, and that its derivative is C^{n-1}
-* or require that, for all m ≤ n, the m-th derivative is continuous, and for all m < n the m-th
-derivative is differentiable.
-The first definition is more efficient for many proofs by induction. The second definition is more
-satisfactory as it gives concrete information about the n-th derivative (contrary to the first point
-of view), and moreover it also makes sense for n = ∞.
-
-Therefore, we give (and use) both definitions, named respectively `times_cont_diff_rec` and
-`times_cont_diff` (as well as relativized versions on a set). We show that they are equivalent.
-The first one is mainly auxiliary: in applications, one should always use `times_cont_diff`
-(but the proofs below use heavily the equivalence to show that `times_cont_diff` is well behaved).
+### Definition of `C^n` functions in domains
+
+One could define `C^n` functions in a domain `s` by fixing an arbitrary choice of derivatives (this
+is what we do with `iterated_fderiv_within`) and requiring that all these derivatives up to `n` are
+continuous. If the derivative is not unique, this could lead to strange behavior like two `C^n`
+functions `f` and `g` on `s` whose sum is not `C^n`. A better definition is thus to say that a
+function is `C^n` inside `s` if it admits a sequence of derivatives up to `n` inside `s`. This
+definition still has the problem that a function which is locally `C^n` would not need to be `C^n`,
+as different choices of sequences of derivatives around different points might possibly not be glued
+together to give a globally defined sequence of derivatives. Also, there are locality problems in
+time: one could image a function which, for each `n`, has a nice sequence of derivatives up to order
+`n`, but they do not coincide for varying `n` and can therefore not be glued to give rise to an
+infinite sequence of derivatives. This would give a function which is `C^n` for all `n`, but not
+`C^∞`. We solve this issue by putting locality conditions in space and time in our definition of
+`times_cont_diff_on`. The resulting definition is slightly more complicated to work with (in fact
+not so much), but it gives rise to completely satisfactory theorems.
+
+### Side of the composition, and universe issues
+
+With a naïve direct definition, the `n`-th derivative of a function belongs to the space
+`E →L[𝕜] (E →L[𝕜] (E ... F)...)))` where there are n iterations of `E →L[𝕜]`. This space
+call also be seen as the space of continuous multilinear functions on `n` copies of `E` with
+values in `F`, by uncurryfication. This is the point of view that is usually adopted in textbooks,
```

Comment:
Wouldn't "uncurrying" be better than "uncurryfication" here? Certainly in normal english usage it would be, but I appreciate the desire to match the names used in the code.

Reply:
I have used `curryfication` both in the code and in the comments because it felt like a natural English word to me, but I may have been using the wrong word all the time. I have just checked with google, and curryfication seems to be a French word :( For instance, https://fr.wikipedia.org/wiki/Curryfication corresponds to https://en.wikipedia.org/wiki/Currying. 

Can you confirm that a sentence such as "by applying them to one variable after the other through curryfication" would be better as "by applying them to one variable after the other through currying"?

Reply:
Replace curryfication with currying everywhere.